desc/__init__.py

```diff
@@ -127,11 +127,15 @@ def set_device(kind="cpu"):
             set_device(kind="cpu")
             return
         devices = [dev for dev in devices if dev["index"] in gpu_ids]
-        for dev in devices:
-            mem = dev["mem_total"] - dev["mem_used"]
-            if mem > maxmem:
-                maxmem = mem
-                selected_gpu = dev
+
+        if (not (gpuid == None)) and (str(gpuid) in gpu_ids):
+            selected_gpu = [dev for dev in devices if dev["index"] == str(gpuid)][0]
```

Comment:
What if the user tries to assign a wrong ID? Shouldn't we throw an error?

Reply:
Thank you for suggesting. If the wrong ID is provided, it should fall back to the previous method: selecting the GPU with maximum free memory. I added a warning message for this case.